cmd/libsnap-confine-private/error.c

```diff
@@ -29,6 +29,8 @@
 static sc_error *sc_error_initv(const char *domain, int code,
 				const char *msgfmt, va_list ap)
 {
+	// Set errno in case we die.
+	errno = 0;
```

Comment:
Shouldn't we save & restore errno before returning, or am I paranoid?

Reply:
It depends on whether the API promises to not modify errno. In this case I would say no, there's a separate call `sc_error_init_from_errno` which takes a copy of errno you want to set in the error. There's a separate SC_ERRNO_DOMAIN domain for errors carrying errno, such that when initializing the error one is expected to pass the errno (copy thereof) in the arguments.

Reply:
I think we can. Note that we plan to change the API of die() to something like die() and die_with_errno() - once that is done this errno = 0; will be removed again.

Reply:
My main concern is obviously cases where sc_error_initv(..) is used to create and initialize sc_error, but after calling it we want to do something extra based on last errno value; and yes @bboozzoo is right in that this is about promise of the API regarding errno modifications. I think it would make sense to make such promise an keep errno untouched on return. 
Since this is going to be changed in followup and errno=0 will be removed I think it's fine as is for this PR.